src/plugins/intel_npu/src/backend/src/zero_executor.cpp

```diff
@@ -49,47 +49,56 @@ ZeroExecutor::ZeroExecutor(const std::shared_ptr<const ZeroInitStructsHolder>& i
                                                       _initStructs->getCommandQueueDdiTable(),
                                                       _config,
                                                       group_ordinal)}} {
-    _logger.debug("ZeroExecutor::ZeroExecutor init start - create graph_command_list");
+    _logger.debug("create graph_command_list");
     OV_ITT_SCOPED_TASK(itt::domains::LevelZeroBackend, "Executor::ZeroExecutor");
     CommandList graph_command_list(_initStructs->getDevice(),
                                    _initStructs->getContext(),
                                    _initStructs->getGraphDdiTable(),
                                    _config,
                                    _group_ordinal);
-    _logger.debug("ZeroExecutor::ZeroExecutor - create graph_command_queue");
+    _logger.debug("create graph_command_queue");
     CommandQueue graph_command_queue(_initStructs->getDevice(),
                                      _initStructs->getContext(),
                                      ZE_COMMAND_QUEUE_PRIORITY_NORMAL,
                                      _initStructs->getCommandQueueDdiTable(),
                                      _config,
                                      _group_ordinal);
-    _logger.debug("ZeroExecutor::ZeroExecutor - create fence");
+    _logger.debug("create fence");
     Fence fence(graph_command_queue, _config);
 
-    _logger.debug("ZeroExecutor::ZeroExecutor - create graph");
     OV_ITT_TASK_CHAIN(ZERO_EXECUTOR_GRAPH, itt::domains::LevelZeroBackend, "Executor::ZeroExecutor", "graphCreate");
 
-    ze_graph_desc_t desc{ZE_STRUCTURE_TYPE_GRAPH_DESC_PROPERTIES,
-                         nullptr,
-                         ZE_GRAPH_FORMAT_NATIVE,
-                         _networkDesc->compiledNetwork.size(),
-                         _networkDesc->compiledNetwork.data(),
-                         nullptr};
-    zeroUtils::throwOnFail(
-        "pfnCreate",
-        _graph_ddi_table_ext->pfnCreate(_initStructs->getContext(), _initStructs->getDevice(), &desc, &_graph));
+    // _graph is a nullptr for CIP path, a new handle will be obtained from the driver based on the given
+    // compiledNetwork _graph gets (reuses) graphHandle from the compiler for CID path
+    if (_networkDesc->metadata.graphHandle == nullptr) {
+        _logger.info("create graph handle on executor");
+        ze_graph_desc_t desc{ZE_STRUCTURE_TYPE_GRAPH_DESC_PROPERTIES,
+                             nullptr,
+                             ZE_GRAPH_FORMAT_NATIVE,
+                             _networkDesc->compiledNetwork.size(),
+                             _networkDesc->compiledNetwork.data(),
+                             nullptr};
+
+        zeroUtils::throwOnFail(
+            "pfnCreate",
+            _graph_ddi_table_ext->pfnCreate(_initStructs->getContext(), _initStructs->getDevice(), &desc, &_graph));
+
+    } else {
+        _logger.info("reuse graph handle created from compiler");
```

Comment:
It looks like debug level logging. The same for the above `_logger.info`

Reply:
thanks, we have change it to `_logger.debug` in the latest commit